Packs/rasterize/Integrations/rasterize/rasterize.py

```diff
@@ -77,449 +66,452 @@ class RasterizeType(Enum):
     JSON = 'json'
 
 
-def check_width_and_height(width: int, height: int) -> tuple[int, int]:
-    """
-    Verifies that the width and height are not greater than the safeguard limit.
-    Args:
-        width: The given width.
-        height: The given height.
+# endregion
 
-    Returns: The checked width and height values - [width, height]
-    """
-    w = min(width, MAX_FULLSCREEN_W)
-    h = min(height, MAX_FULLSCREEN_H)
+# region utility classes
 
-    return w, h
+def excepthook_recv_loop(args):
+    exc_value = args.exc_value
+    if args.exc_type == json.decoder.JSONDecodeError:
+        demisto.debug(f"Caught a JSONDecodeError exception in _recv_loop, suppressing, {exc_value}")
+    else:
+        demisto.info(f"Unsuppressed Exception in _recv_loop: {args.exc_type=}, {exc_value=}")
 
 
-def return_err_or_warn(msg):
-    return_error(msg) if WITH_ERRORS else return_warning(msg, exit=True)
+class TabLifecycleManager:
+    def __init__(self, browser, chrome_port, offline_mode):
+        self.browser = browser
+        self.chrome_port = chrome_port
+        self.offline_mode = offline_mode
+        self.tab = None
 
+        demisto.debug(f'TabLifecycleManager, __init__, {self.chrome_port=}, active tabs len: {len(self.browser.list_tab())}')
 
-def opt_name(opt):
-    return opt.split('=', 1)[0]
+    def __enter__(self):
+        self.tab = self.browser.new_tab()
 
+        self.tab.start()
+        if self.offline_mode:
+            self.tab.Network.emulateNetworkConditions(offline=True, latency=-1, downloadThroughput=-1, uploadThroughput=-1)
```

Comment:
[nit] magic numbers + I'm assuming -1 has a special meaning?

Reply:
Yes, that's how it's done. Not exactly magic numbers.